File: src/main/Game.java
package main;

import gamestate.GameStateMachine;
import gui.GuiManager;
import handlers.Resources;
import org.newdawn.slick.BasicGame;
import org.newdawn.slick.GameContainer;
import org.newdawn.slick.Graphics;
import org.newdawn.slick.SlickException;

public class Game extends BasicGame {
	
	public Game(String title) {
		super(title);
	}
	
	public void render(GameContainer gc, Graphics g) throws SlickException {
		if (!Resources.isReady()) {
			Resources.getSplashImage().draw();
		} else {
			GameStateMachine.INSTANCE.render(gc, g);
		}
	}
	
	public void init(GameContainer gc) throws SlickException {
		Resources.init();
	}
	
	public void update(GameContainer gc, int delta) throws SlickException {
		if (!Resources.isReady()) {
			Resources.initPool();
			GameStateMachine.INSTANCE.init(gc);
			GuiManager.init(gc);
			Resources.setReady(true);
		} else {
			GameStateMachine.INSTANCE.update(gc);
		}
		Main.ticks++;
	}
	
}

File: src/main/Game.kt
package main

import gamestate.GameStateMachine
import gui.GuiManager
import util.Resources
import org.newdawn.slick.BasicGame
import org.newdawn.slick.GameContainer
import org.newdawn.slick.Graphics
import org.newdawn.slick.SlickException

class Game(title: String) : BasicGame(title) {

    override fun render(gc: GameContainer, g: Graphics) {
        if (!Resources.isReady) {
            Resources.splashImage.draw()
        } else {
            GameStateMachine.render(gc, g)
        }
    }

    override fun init(gc: GameContainer) {
        Resources.init()
    }

    override fun update(gc: GameContainer, delta: Int) {
        if (!Resources.isReady) {
            Resources.initPool()
            GameStateMachine.init(gc)
            GuiManager.init(gc)
            Resources.isReady = true
        } else {
            GameStateMachine.update(gc)
        }
        Main.ticks++
    }

}
